pkg/controller/statusmanager/pod_status.go

```diff
@@ -175,54 +205,72 @@ func (status *StatusManager) updateNodeStatus(nodeName string, ready bool, reaso
-			log.Error(err, "Failed to get nodes for condition updating")
+			log.Error(err, fmt.Sprintf("Error getting %s for node condition", operatortypes.NsxNodeAgentContainerName))
 			return
 		}
-		for _, node := range nodes.Items {
-			status.setNodeNetworkUnavailable(node.ObjectMeta.Name, false, "NSXNodeAgent", "Waiting for nsx-node-agent to be created", sharedInfo)
-		}
-		return
 	}
-	pod := corev1.Pod{}
-	for _, pod = range pods.Items {
-		containerStatus := corev1.ContainerStatus{}
-		ready := true
-		messages := ""
-		nodeName := pod.Spec.NodeName
-		for _, containerStatus = range pod.Status.ContainerStatuses {
-			if containerStatus.State.Running == nil {
-				ready = false
-				if containerStatus.State.Waiting != nil {
-					if containerStatus.State.Waiting.Message == "" {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Waiting.Reason)
-					} else {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Waiting.Message)
-					}
-				} else if containerStatus.State.Terminated != nil {
-					if containerStatus.State.Terminated.Message == "" {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Terminated.Reason)
-					} else {
-						messages = messages + fmt.Sprintf("nsx-node-agent/%s not running: %s. ", containerStatus.Name, containerStatus.State.Terminated.Message)
-					}
+	containerStatus := corev1.ContainerStatus{}
+	ready := true
+	// messages is used to store the error message from container status,
+	// tmpMsgs is for the status that pod just restarted in a short time
+	var messages, tmpMsgs string
+	nodeName := pod.Spec.NodeName
+	var startedAt *time.Time
+	var lastStartedAt *time.Time
+	// when pod is pending, its ContainerStatuses is empty
+	if len(pod.Status.ContainerStatuses) == 0 {
+		ready = false
+		messages = fmt.Sprintf("nsx-node-agent not running: %s.", pod.Status.Phase)
+	}
+	for _, containerStatus = range pod.Status.ContainerStatuses {
+		if containerStatus.State.Running == nil {
+			ready = false
+			if containerStatus.State.Waiting != nil {
+				if containerStatus.State.Waiting.Message == "" {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Waiting.Reason)
+				} else {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Waiting.Message)
 				}
-
+			} else if containerStatus.State.Terminated != nil {
+				if containerStatus.State.Terminated.Message == "" {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Terminated.Reason)
+				} else {
+					messages = messages + fmt.Sprintf("%s/%s not running: %s. ", pod.Name, containerStatus.Name, containerStatus.State.Terminated.Message)
+				}
+			}
```

Comment:
what is the last else case here? could there be anything?

Reply:
Good point. In https://godoc.org/k8s.io/api/core/v1#ContainerState , there're only 3 state Waiting, Running, Terminated. So I think there won't be another case.